jpa/subsystem/src/main/java/org/jboss/as/jpa/config/PersistenceUnitMetadataImpl.java

```diff
@@ -379,6 +379,11 @@ public boolean needsJPADelegatingClassFileTransformer() {
 
     @Override
     public void addTransformer(ClassTransformer classTransformer) {
+        // Do not add Hibernate ORM 6.x/7.x bytecode enhancement class transformers
+        // (e.g. org.hibernate.jpa.internal.enhance.EnhancingClassTransformerImpl)
+        if(classTransformer.getClass().getName().startsWith("org.hibernate")) {
```

Comment:
Should have mentioned in the first review, but if you want to check for the package prefix, the string to check for should really be `""org.hibernate."` - notice the trailing dot. Because `org.hibernate` would match something like `org.hibernateX`.

Reply:
Good catch.